test/regex_test.cpp

```diff
@@ -0,0 +1,186 @@
+#include "gtest/gtest.h"
+#include "regex_yaml.h"
+#include "stream.h"
+
+using YAML::RegEx;
+using YAML::Stream;
+
+namespace {
+TEST(RegExTest, Empty) {
+  RegEx empty;
+  EXPECT_TRUE(empty.Matches(std::string()));
+  EXPECT_EQ(0, empty.Match(std::string()));
+  for (int ii = Stream::eof() + 1; ii < 128; ++ii) {
+    EXPECT_FALSE(empty.Matches(char(ii)));
+    EXPECT_EQ(-1, empty.Match(char(ii)));
+  }
+}
+
+TEST(RegExTest, SingleCharacter) {
+  for (int ii = Stream::eof() + 1; ii < 128; ++ii) {
+    RegEx ex((char)ii);
+    for (int jj = Stream::eof() + 1; jj < 128; ++jj) {
+      if (jj == ii) {
+        EXPECT_TRUE(ex.Matches(char(jj)));
+        EXPECT_EQ(1, ex.Match(char(jj)));
+
+        // Match at start of string only!
+        std::string prefixed = std::string(1, ii + 1) +
+                               std::string("prefix: ") +
+                               std::string(1, char(jj));
+        EXPECT_FALSE(ex.Matches(prefixed));
+        EXPECT_EQ(-1, ex.Match(prefixed));
+      } else {
+        EXPECT_FALSE(ex.Matches(char(jj)));
+        EXPECT_EQ(-1, ex.Match(char(jj)));
+      }
+    }
+  }
+}
+
+TEST(RegExTest, Range) {
+  for (int ii = Stream::eof() + 1; ii < 127; ++ii) {
+    for (int jj = Stream::eof() + 1; jj < 128; ++jj) {
+      RegEx ex((char)ii, (char)jj);
+      for (int kk = Stream::eof() + 1; kk < 128; ++kk) {
+        if (ii <= kk && kk <= jj) {
+          EXPECT_TRUE(ex.Matches(char(kk)));
+          EXPECT_EQ(1, ex.Match(char(kk)));
+        } else {
+          EXPECT_FALSE(ex.Matches(char(kk)));
+          EXPECT_EQ(-1, ex.Match(char(kk)));
+        }
+      }
+    }
+  }
+}
+
+TEST(RegExTest, EmptyString) {
+  RegEx ex = RegEx(std::string());
+  EXPECT_TRUE(ex.Matches(std::string()));
+  EXPECT_EQ(0, ex.Match(std::string()));
+
+  // Matches anything, unlike RegEx()!
+  EXPECT_TRUE(ex.Matches(std::string("hello")));
+  EXPECT_EQ(0, ex.Match(std::string("hello")));
+}
+
+TEST(RegExTest, SingleCharacterString) {
```

Comment:
If you're testing this, what's the point in testing the SingleCharacter, given that you even had to add to the public API in order to test individual characters?